2024/incoming/perl-version-bumper.pod

```diff
@@ -0,0 +1,619 @@
+would point to likely errors in programming (like undeclared variables
+or undefined values).
+
+Since Perl v5.36, these two statements are implicitely included inr
+C<use VERSION>. It also enables the official features of that version
+of the language, and disables the deprecated ones.
+
+Therefore, C<use VERSION> helps define a good I<baseline> for the Perl
+language, so that the compiler can fully understand the code that follows.
+It's of course possible to include experimental features, or re-enable
+deprecated feature (the latter is really not recommended), to fine-tune
+the specific dialect of Perl that the code is written in.
+
+Always use a I<stable> version number with C<use VERSION>. The point is to
+I<declare> which version of the Perl I<language> your code is targetting.
+
+Future versions of C<perl> implicitely promise that they will understand
+the dialect of Perl declared by C<use VERSION>, and that they will run
+it to the best of their abilities.
+
+=head2 The Once and Future Perl
+
+Every year, a new version of C<perl> comes out. The C<perl> binary is
+updated, and so is the Perl language. The Perl 5 Porters support the
+two most recent stable releases of Perl, which gives you a good reason
+to upgrade your binary. You don't have to upgrade the version of the
+language your code uses, though.
+
+If you use Perl in 2024, you don't have to write code like the dot-com
+bubble is about to burst. (Be wary of the AI bubble, though...)
+You can easily signal your code was written in the last decade,
+and enjoy the recent evolutions of the language.
+
+=head3 The new "use strict and warnings"
+
+Since the time of "use strict and warnings" as minimum requirements for
+decent Perl code, the language has evolved, and brought in more useful
+features. It also deprecated ancient features that were deemed bad ideas
+in a modern context (such as the indirect object notation).
+
+Writing C<use VERSION> as the first (active) line of any Perl code
+will I<declare> to the C<perl> interpreter (and to whoever is reading
+the code) which version of the Perl I<language> the code that follows
+is written in.
+
+So, just put a C<use VERSION> line at the top of all your Perl files.
+
+Whenever a new stable version is released (which happens every year
+towards the end of May), you should at least read the L<perldelta> that
+accompanies it, and check if you see some new feature you think you'd
+want to use.
+
+There might exist some compatibility modules that you can use,
+to get a taste of those new features on version of C<perl>
+that doesn't support them natively. They might behave slightly
+differently, though.
+
+C<v5.36> (released in May 2022) contains a very good mix of
+stable features (default C<strict> and C<warnings>, C<signatures>,
+C<isa>), as well as the removal of deprecated features (C<indirect>,
+C<multidimensional>).
+
+In 2024, the main Linux distributions all include a version of Perl that
+will support C<use v5.36;>.
```

Comment:
Arguably, but who decides which distros are "main"? The stable version distro I use is currently shipping with 5.34 (and I'm perfectly happy with that :-)

Instead of "the main", perhaps just go with "many" or even "most"?

Reply:
It never occured to me that the box I wrote this entire article on sports a Perl v5.32... 😅

Reply:
```suggestion
Most Linux distributions released in 2024 include a version of Perl that
will support C<use v5.36;>.
```